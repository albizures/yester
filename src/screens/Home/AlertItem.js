import React from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet, TouchableOpacity } from 'react-native'
import colors from '../../utils/colors'
import { Title, Heading3 } from '../../components'
import { subscriptionStatus } from '../../utils/session'

const AlertItem = (props) => {
  const { currentStatus, onPress } = props

  const conditionalText = {
    [subscriptionStatus.ODD_REQUIRE.code]: {
      message: 'home.alertItem.message.trial',
    },
    [subscriptionStatus.EVEN_REQUIRE.code]: {
      message: 'home.alertItem.message.trial',
    },
    [subscriptionStatus.EXPIRED.code]: {
      message: 'home.alertItem.message.expired',
    },
  }

  const text = conditionalText[currentStatus.code]

  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <View style={styles.card}>
        <View style={styles.topView}>
          <Title keyName='home.alertItem.title' style={styles.dayTopicText} />
        </View>
        <View style={styles.bottomView}>
          <View style={{ flex: 1 }}>
            <Heading3 keyName={text.message} numberOfLines={2} style={{ flexWrap: 'wrap' }} />
          </View>
        </View>
      </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    marginBottom: 20,
    marginHorizontal: 20,
  },
  card: {
    backgroundColor: colors.white,
    borderRadius: 10,
    borderWidth: 0.5,
    borderColor: colors.white,
    shadowColor: colors.shadow,
    shadowOpacity: 0.22,
    shadowRadius: 15,
    shadowOffset: {
      height: 8,
    },
  },
  topView: {
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    paddingHorizontal: 20,
    paddingVertical: 16,
    backgroundColor: colors.bittersweet,
  },
  bottomView: {
    flexDirection: 'row',
    padding: 10,
    paddingTop: 5,
    alignItems: 'stretch',
  },
  newTextContainer: {
    marginTop: 10,
    marginHorizontal: 10,
  },
  newText: {
    color: colors.mineShaft,
    fontWeight: 'bold',
  },
  dayTopicText: {
    color: colors.white,
    fontWeight: 'bold',
  },
})

AlertItem.propTypes = {
  currentStatus: PropTypes.object.isRequired,
  onPress: PropTypes.func.isRequired,
}
export default AlertItem
