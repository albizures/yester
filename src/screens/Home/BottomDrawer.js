import React from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet, Image, Dimensions, Text } from 'react-native'
import Modal from 'react-native-modal'
import colors from '../../utils/colors'
import icons from '../../utils/icons'
import { Title, Heading5, Body1 } from '../../components'
import Button from '../../components/Button'
import Divider from '../../components/Divider'
import { subscriptionStatus } from '../../utils/session'

const BottomDrawer = ({ props, visible, onOk, onSubscribe }) => {
  const {
    contextUser: { currentStatus, stats, user },
  } = props
  const { givenName } = user

  const completed = stats.storyCounter ? Math.round((stats.storyCounter / 305) * 100) : 0

  const showLeft = currentStatus === subscriptionStatus.PREVIEW
  let left = 5 - stats.storyCounter
  left = left < 0 ? 0 : left

  let actions = (
    <View style={{ width: '100%' }}>
      <Title style={styles.secondTitle} keyName='home.bottomDrawer.secondTitle' />
      <View style={styles.statsRow}>
        <Image source={icons.pieChart} style={{ width: 22, height: 22, marginRight: 12 }} />
        <Text style={{ textAlignVertical: 'top', marginRight: 23 }}>
          <Heading5 keyName='home.bottomDrawer.stories1' />
          <Heading5
            style={{ fontWeight: 'bold' }}
            keyName='home.bottomDrawer.stories2'
            data={{ storyCounter: stats.storyCounter }}
          />
          <Heading5 keyName='home.bottomDrawer.stories3' />
          <Heading5
            style={{ fontWeight: 'bold' }}
            keyName='home.bottomDrawer.stories4'
            data={{ completed }}
          />
          <Heading5 keyName='home.bottomDrawer.stories5' />
        </Text>
      </View>
      <View style={styles.actionsRow}>
        <Body1 style={styles.secondAction} text=' ' onPress={onOk} />
        <Button title='home.bottomDrawer.button.ok' onPress={onOk} style={styles.button} />
      </View>
    </View>
  )
  if (showLeft) {
    actions = (
      <View style={{ width: '100%' }}>
        <View style={styles.statsRow}>
          <Image source={icons.alertCircle} style={{ width: 22, height: 22, marginRight: 12 }} />
          <Heading5
            style={{ fontStyle: 'italic', textAlignVertical: 'top', marginRight: 23 }}
            keyName='home.bottomDrawer.storiesLeft'
            data={{ left: left }}
          />
        </View>
        <View style={styles.actionsRow}>
          <Body1 style={styles.secondAction} keyName='home.bottomDrawer.dismiss' onPress={onOk} />
          <Button
            title='home.bottomDrawer.button.subscribe'
            onPress={onSubscribe}
            style={styles.button}
          />
        </View>
      </View>
    )
  }

  return (
    <Modal
      isVisible={visible}
      backdropColor='transparent'
      swipeDirection='down'
      onSwipeComplete={onOk}
      onBackButtonPress={onOk}
      onBackdropPress={onOk}
      animationInTiming={500}
      animationOutTiming={500}
      style={{ margin: 0 }}
    >
      <View style={styles.modalContainer}>
        <View style={styles.card}>
          <View style={styles.container}>
            <View style={styles.divider} />
            <Title
              style={styles.title}
              keyName='home.bottomDrawer.title'
              data={{ name: givenName }}
            />
            <Heading5 style={{ marginBottom: 33 }} keyName='home.bottomDrawer.message' />

            <Divider style={{ width: 323, marginBottom: 18 }} />

            {actions}
          </View>
        </View>
      </View>
    </Modal>
  )
}

BottomDrawer.propTypes = {
  props: PropTypes.object.isRequired,
  contextUser: PropTypes.object.isRequired,
  visible: PropTypes.bool.isRequired,
  onOk: PropTypes.func.isRequired,
  onSubscribe: PropTypes.func.isRequired,
}

export default BottomDrawer

const { width, height } = Dimensions.get('window')
const styles = StyleSheet.create({
  modalContainer: {
    width,
    height,
    justifyContent: 'flex-end',
  },
  card: {
    width: '100%',
    height: 360,
    // overflow: 'hidden',
    borderTopStartRadius: 30,
    borderTopEndRadius: 30,
    borderWidth: 0.5,
    backgroundColor: colors.white,
    borderColor: colors.white,
    shadowColor: colors.questionCardShadow,
    shadowOpacity: 0.1,
    shadowRadius: 30,
    shadowOffset: {
      height: 30,
    },
  },
  container: {
    width: '100%',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    paddingTop: 12,
    paddingHorizontal: 23,
  },
  divider: {
    width: 97,
    height: 3,
    alignSelf: 'center',
    borderWidth: 0,
    borderRadius: 1.5,
    borderColor: colors.mischka,
    backgroundColor: colors.mischka,
  },
  title: {
    fontWeight: 'bold',
    marginTop: 33,
    marginBottom: 13,
  },
  secondTitle: {
    color: colors.governorBay,
    fontWeight: 'bold',
    marginBottom: 13,
  },
  statsRow: {
    maxWidth: '100%',
    flexDirection: 'row',
    marginRight: 23,
  },
  actionsRow: {
    width: '100%',
    height: 50,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: 20,
    marginBottom: 27,
  },
  button: { width: 150 },
  secondAction: {
    textDecorationLine: 'underline',
    color: colors.governorBay,
  },
})
