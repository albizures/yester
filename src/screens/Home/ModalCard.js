import React from 'react'
import PropTypes from 'prop-types'
import { View, Image, StyleSheet, TouchableOpacity, Dimensions } from 'react-native'

import colors from '../../utils/colors'
import { getCategoryIllustration } from '../../utils/icons'
import { capitalize } from '../../utils'

import Button, {types} from '../../components/Button'
import { Heading2, Heading5, Heading3 } from '../../components'
import withAges, { shapeContextAges } from '../../components/withAges'

class ModalCard extends React.Component {
  static propTypes = {
    contextAges: PropTypes.shape(shapeContextAges).isRequired,
    navigation: PropTypes.object.isRequired,
  }

  onWrite = () => {
    const { navigation } = this.props

    const ageId = navigation.getParam('ageId')
    const questionId = navigation.getParam('questionId')
    const question = navigation.getParam('question')
    const storyId = navigation.getParam('storyId')

    navigation.navigate('Writing', { ageId, questionId, question, storyId })
  }

  onSkip = () => {
    const { navigation } = this.props
    navigation.goBack()
  }

  render () {
    const { navigation } = this.props
    const { ages } = this.props.contextAges
    const ageId = navigation.getParam('ageId')
    const question = navigation.getParam('question', '')
    const categoryId = navigation.getParam('categoryId')
    const age = (ages[ageId] || {}).name
    const illustration = getCategoryIllustration(categoryId)

    const ComponentQuestion = question.length > 65 ? Heading3 : Heading2
    return (
      <View style={styles.modalContainer}>
        <TouchableOpacity style={styles.modalBackground} onPress={this.onSkip} >
          <View />
        </TouchableOpacity>
        <View style={styles.card}>
          <View style={styles.container}>
            <Image source={illustration} style={{width: 340, height: 250}} />
            <View style={{flex: 1, position: 'absolute', paddingTop: 27}}>
              <Heading5 text={capitalize(age)} style={{textAlign: 'center'}} />
            </View>

            <View style={styles.contentTop}>
              {/* <Heading2 text={capitalize(category)} style={{marginBottom: 10}} /> */}
              <ComponentQuestion style={{textAlign: 'center'}} text={capitalize(question)} />
            </View>

            <View style={styles.contentBottom}>
              <Button title='questionCard.write' onPress={this.onWrite} style={{marginBottom: 20}} />
              <Button title='questionCard.skip' onPress={this.onSkip} type={types.OUTLINED} />
            </View>
          </View>
        </View>
      </View>
    )
  }
}

export default withAges(ModalCard)

const { width, height } = Dimensions.get('window')
const styles = StyleSheet.create({
  modalBackground: {
    flex: 1,
    backgroundColor: 'rgba(98, 97, 232, 0.85)',
    width: '100%',
  },
  modalContainer: {
    flex: 1,
    width,
    height,
    alignItems: 'center',
    justifyContent: 'center',
  },
  card: {
    position: 'absolute',
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden',
    borderRadius: 30,
    borderWidth: 0.5,
    margin: 20,
    borderColor: colors.white,
    shadowColor: colors.questionCardShadow,
    shadowOpacity: 0.1,
    shadowRadius: 30,
    shadowOffset: {
      height: 30,
    },
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-start',
    backgroundColor: colors.white,
  },
  contentTop: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-start',
    paddingHorizontal: 20,
    paddingBottom: 20,
  },
  contentBottom: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-end',
    paddingBottom: 30,
  },
})
