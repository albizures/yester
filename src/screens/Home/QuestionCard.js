import React from 'react'
import PropTypes from 'prop-types'
import { View, Image, StyleSheet } from 'react-native'
import colors from '../../utils/colors'
import icons from '../../utils/icons'
import Button, {types} from '../../components/Button'
import { Heading2, Heading5 } from '../../components'

const QuestionCard = (props) => {
  const {data, onPressWrite, onPressSkip} = props

  return (
    <View style={styles.card}>
      <View style={styles.container}>
        <View style={styles.ilustration}>
          <Image source={icons.ssCard}
            style={{width: 340, height: 250}} />
          <View style={{flex: 1, position: 'absolute', alignSelf: 'center', paddingTop: 27}}>
            <Heading5 text={data.age} style={[{textAlign: 'center'}]} />
          </View>
        </View>
        <View style={styles.content}>
          <Heading2 text={data.category} style={[{marginBottom: 10}]} />
          <View style={{height: 41}}>
            <Heading5 text={data.text} />
          </View>
          <Button title='questionCard.write'
            style={{marginVertical: 20}}
            onPress={onPressWrite} />
          <Button title='questionCard.skip' onPress={onPressSkip} type={types.OUTLINED} />
        </View>
      </View>
    </View>
  )
}

QuestionCard.propTypes = {
  data: PropTypes.object.isRequired,
  onPressWrite: PropTypes.func.isRequired,
  onPressSkip: PropTypes.func.isRequired,
}

export default QuestionCard

const styles = StyleSheet.create({
  card: {
    height: 520,
    width: 340,
    borderRadius: 30,
    borderWidth: 0.5,
    borderColor: colors.white,
    marginTop: 84,
    shadowColor: colors.questionCardShadow,
    shadowOpacity: 0.1,
    shadowRadius: 30,
    shadowOffset: {
      height: 30,
    },
    overflow: 'hidden',
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
    backgroundColor: colors.white,
    borderRadius: 30,
  },
  ilustration: {
    height: 250,
  },
  content: {
    height: 270,
    alignItems: 'center',
    justifyContent: 'space-around',
    paddingHorizontal: 20,
    paddingTop: 20,
    paddingBottom: 30,
  },
})
