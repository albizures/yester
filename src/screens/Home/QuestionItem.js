import React from 'react'
import PropTypes from 'prop-types'
import { View, Image, StyleSheet, TouchableOpacity } from 'react-native'
import colors from '../../utils/colors'
import icons from '../../utils/icons'
import { Title, Heading3, Description } from '../../components'
import IconButton from '../../components/IconButton'
import { capitalize } from '../../utils'

const QuestionItem = (props) => {
  const {data, onPress} = props

  return (
    <TouchableOpacity onPress={onPress} style={styles.container} >
      <View style={styles.card}>
        <View style={styles.topView}>
          <Title text={'Your topic of the day!'} style={styles.dayTopicText} />
        </View>
        <View style={styles.bottomView}>
          <View style={styles.textView}>
            <Description keyName='home.questionItem.new' style={styles.newText} />
            <Heading3 text={data.text} numberOfLines={2} style={{flexWrap: 'wrap'}} />
          </View>
          <View style={styles.iconView}>
            <IconButton>
              <Image source={icons.pencil} style={styles.icon} />
            </IconButton>
          </View>
        </View>

      </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    height: 138,
    width: 340,
  },
  card: {
    flex: 1,
    backgroundColor: colors.white,
    borderRadius: 10,
    borderWidth: 0.5,
    borderColor: colors.white,
    shadowColor: colors.shadow,
    shadowOpacity: 0.22,
    shadowRadius: 15,
    shadowOffset: {
      height: 8,
    },
  },
  topView: {
    height: 58,
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    alignItems: 'flex-start',
    justifyContent: 'center',
    backgroundColor: colors.royalBlue,
    paddingLeft: 20,
  },
  bottomView: {
    flex: 1,
    flexDirection: 'row',
    paddingLeft: 20,
  },
  textView: {
    flex: 1,
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  },
  iconView: {
    width: 50,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 15,
  },
  icon: {
    width: 17,
    height: 17,
  },
  newText: {
    color: colors.royalBlue,
    fontWeight: 'bold',
    marginTop: 10,
    marginBottom: 7,
  },
  dayTopicText: {
    color: colors.white,
    fontWeight: 'bold',
  },
})

QuestionItem.propTypes = {
  data: PropTypes.object.isRequired,
  onPress: PropTypes.func.isRequired,
}
export default QuestionItem
