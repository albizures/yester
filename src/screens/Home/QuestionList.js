import React from 'react'
import PropTypes from 'prop-types'
import { Text, View, FlatList, StyleSheet } from 'react-native'

const QuestionList = (props) => (

  <FlatList
    data={props.ages}
    renderItem={({item}) => (
      <View>
        <Text style={localStyles.sectionHeader}>{item.text}</Text>
        <TopicList {...props} />
      </View>
    )}
    keyExtractor={keyExtractor}
  />
)

QuestionList.propTypes = {
  ages: PropTypes.array.isRequired,
  topics: PropTypes.array.isRequired,
}

export default QuestionList

const keyExtractor = (item, index) => {
  return index.toString()
}

const TopicList = (props) => (
  <FlatList
    data={props.topics}
    renderItem={({item}) => (
      <Text style={localStyles.item}>{item.text}</Text>
    )}
    keyExtractor={keyExtractor}
  />
)

TopicList.propTypes = {
  topics: PropTypes.array.isRequired,
}

const localStyles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 22,
  },
  sectionHeader: {
    paddingTop: 2,
    paddingLeft: 10,
    paddingRight: 10,
    paddingBottom: 2,
    fontSize: 14,
    fontWeight: 'bold',
    backgroundColor: 'rgba(247,247,247,1.0)',
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
})
