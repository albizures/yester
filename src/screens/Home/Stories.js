import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet, FlatList, Alert, Image, Dimensions, Animated, Text } from 'react-native'
import QuestionItem from './QuestionItem'
import StoryItem from './StoryItem'
import Tabs from './Tabs'
import Container from '../../components/Container'
import TopBar from '../../components/TopBar'
import { Title, Heading4, Heading3 } from '../../components'
import http from '../../utils/http'
import colors from '../../utils/colors'
import icons from '../../utils/icons'
import { indexToString, capitalize } from '../../utils'
import { translate } from '../../components/Translate'
import { screen, track } from '../../utils/analytics'
import debugFactory from 'debug'

const debugInfo = debugFactory('yester:Stories:info')
const debugError = debugFactory('yester:Stories:error')

export default class Stories extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
  }

  state = {
    isLoading: true,
    item: {},
    positionToast: new Animated.Value(-100),
  }

  willFocusListener = null

  async componentDidMount () {
    const { navigation } = this.props
    const { addListener } = navigation
    this.willFocusListener = addListener('willFocus', this.load)

    /*
    this.setState({
      isLoading: true,
    })
    try {
      const { data: question } = await http.getAPI('/v2/questions')
      this.setState({ question })
    } catch (error) {
      debugError('today question:', error.response)
      if (error.response.status !== 404) {
        Alert.alert(translate('home.error.today.question'))
      }
    }
*/
    this.setState({ isLoading: false })

    const { positionToast } = this.state
    const storyId = navigation.getParam('storyId')
    if (storyId) {
      Animated.spring(positionToast, {
        toValue: 40,
        bounciness: 3,
        speed: 3,
      }).start()

      this.timeout = setTimeout(this.closeToast, 5000)
    }
  }

  load = () => {
    screen('My Story', { age: 'childhood' })
  }

  onPressToast = () => {
    const { navigation } = this.props
    const storyId = navigation.getParam('storyId')
    this.closeToast()
    navigation.navigate('Reading', { storyId })
  }

  closeToast = () => {
    const { positionToast } = this.state
    clearInterval(this.timeout)
    Animated.spring(positionToast, {
      toValue: -100,
      bounciness: 3,
      speed: 3,
    }).start()
  }

  componentWillUnmount () {
    clearInterval(this.timeout)
    this.willFocusListener.remove()
  }

  renderChapter = ({ item }) => {
    const { questions } = this.state
    return (
      <View style={styles.chapterView}>
        <View style={styles.chapterTitle}>
          <Image source={icons.childhood} style={styles.chapterImage} />
          <Title text={capitalize(item.text)} style={styles.chapterText} />
        </View>
        <FlatList data={questions} renderItem={this.renderStoryItem} keyExtractor={indexToString} />
      </View>
    )
  }

  renderStoryItem = ({ item }) => <StoryItem data={item} onPress={() => this.onPressItem(item)} />

  onWriteTodayQuestion = () => {
    const { question: item } = this.state
    const {
      category,
      age_id: ageId,
      description: question,
      id: questionId,
      story_id: storyId,
      category_id: categoryId,
    } = item
    this.onPressItem({
      ageId,
      category,
      question,
      questionId,
      storyId,
      categoryId,
    })
  }

  onPressItem = (item) => {
    const { navigation } = this.props
    const { ageId, category, question, questionId, storyId, content, categoryId } = item

    track('Tap Story', { ageId, category, title: question })

    if (content) {
      return navigation.navigate('Reading', { storyId })
    }

    debugInfo(item)

    navigation.navigate('ModalCard', {
      ageId,
      category,
      question,
      questionId,
      storyId,
      categoryId,
    })
  }

  render () {
    const { isLoading, question, positionToast } = this.state
    const topBarTitle = (
      <View style={{ height: 51, alignItems: 'center', justifyContent: 'center' }}>
        <Image source={icons.logoWhite} style={styles.topBarImage} />
      </View>
    )
    const topBar = <TopBar title={topBarTitle} />
    return (
      <Container topBar={topBar} isLoading={isLoading} style={styles.container}>
        <View style={styles.view}>
          {question && (
            <QuestionItem text={question.description} onPress={this.onWriteTodayQuestion} />
          )}
          {!isLoading && <Tabs onPressItem={this.onPressItem} answered />}
        </View>
        <Animated.View style={[styles.toastContainer, { bottom: positionToast }]}>
          <View style={[styles.toast]}>
            <Text style={styles.checkMark}>✓</Text>
            <View style={styles.contentToast}>
              <Heading4 keyName='home.toast.story.saved' />
              <Heading3
                keyName='home.toast.read.story.now'
                onPress={this.onPressToast}
                style={{ textDecorationLine: 'underline' }}
              />
            </View>
            <View style={styles.closeContainer}>
              <Text style={styles.close} onPress={this.closeToast}>
                ×
              </Text>
            </View>
          </View>
        </Animated.View>
      </Container>
    )
  }
}

const { width } = Dimensions.get('window')
const styles = StyleSheet.create({
  toastContainer: {
    position: 'absolute',
    shadowColor: colors.shadow,
    shadowOpacity: 0.3,
    shadowRadius: 8,
    shadowOffset: {
      height: 8,
    },
    elevation: 8,
  },
  toast: {
    flexDirection: 'row',
    backgroundColor: colors.white,
    fontSize: 30,
    borderRadius: 10,
    overflow: 'hidden',
  },
  contentToast: {
    paddingHorizontal: 15,
    paddingVertical: 10,
    height: 59,
  },
  closeContainer: {
    marginVertical: 10,
    paddingHorizontal: 14,
    borderLeftWidth: 1,
    borderLeftColor: colors.mischka,
    height: 40,
  },
  close: {
    fontSize: 30,
    lineHeight: 40,
    color: colors.boulder,
    textAlign: 'center',
    fontWeight: 'bold',
  },
  checkMark: {
    paddingHorizontal: 15,
    paddingVertical: 5,
    height: 60,
    width: 60,
    backgroundColor: colors.mantis,
    fontSize: 40,
    color: colors.white,
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-start',
    backgroundColor: colors.athensGray,
  },
  topBarImage: {
    width: 100,
    height: 26,
  },
  view: {
    flex: 1,
    backgroundColor: colors.athensGray,
    paddingTop: 20,
  },
  chapterView: {
    width,
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  chapterTitle: {
    width,
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    paddingLeft: 40,
    marginTop: 33,
    marginBottom: 20,
  },
  chapterImage: {
    width: 28,
    height: 30,
    marginRight: 10,
  },
  chapterText: {
    color: colors.governorBay,
    fontWeight: 'bold',
    paddingTop: 5,
  },
})
