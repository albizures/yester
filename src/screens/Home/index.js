import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet, FlatList, Alert, Image, Modal } from 'react-native'
import QuestionItem from './QuestionItem'
import QuestionCard from './QuestionCard'
import Container from '../../components/Container'
import TopBar from '../../components/TopBar'
import { Heading2, Title } from '../../components'
import http from '../../utils/http'
import colors from '../../utils/colors'
import icons from '../../utils/icons'
import { indexToString, capitalize } from '../../utils'

export default class Home extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
  }

  state = {
    isLoading: true,
    modalVisible: false,
    item: {},
  }

  setModalVisible (visible, item) {
    this.setState({modalVisible: visible, item: item})
  }

  async componentDidMount () {
    this.setState({
      isLoading: true,
    })
    try {
      const { data } = await http.get('https://my-json-server.typicode.com/gluix20/treasure/db')
      this.setState({
        isLoading: false,
        ...data,
      })
    } catch (error) {
      Alert.alert('Error', error.message)
    }
  }

  renderAgeItem = ({item}) => (
    <View>
      <Heading2 text={capitalize(item.text)} style={[styles.ageText]} />
      <Image source={icons.ssSonaja}
        style={{width: 60, height: 64.52, alignSelf: 'center', marginBottom: 23}} />
      <View style={{flex: 1, alignItems: 'center'}}>
        {this.getTopicsItems()}
      </View>
    </View>
  )

  renderTopicItem = ({item}) => (
    <QuestionItem
      data={item}
      onPress={() => this.onPressItem(item)}
    />
  )

  getAgesItems = () => (
    <FlatList
      data={this.state.ages}
      renderItem={this.renderAgeItem}
      keyExtractor={indexToString}
    />
  )

  getTopicsItems = () => (
    <FlatList
      data={this.state.questions}
      renderItem={this.renderTopicItem}
      keyExtractor={indexToString}
    />
  )

  onPressItem = (item) => {
    this.setModalVisible(true, item)
  }

  onPressWrite = () => {
    this.setModalVisible(!this.state.modalVisible)
    this.props.navigation.navigate('Writing')
  }

  onPressSkip = () => {
    this.setModalVisible(!this.state.modalVisible)
  }

  render () {
    const { isLoading } = this.state
    const topBarTitle = (
      <View style={{flex: 1, flexDirection: 'row', justifyContent: 'center'}}>
        <Image style={{ height: 25, width: 20 }} source={require('../../assets/feather-white.png')} />
        <Title keyName='home.title' style={{color: colors.white}} />
      </View>
    )

    const topBar = (
      <TopBar title={topBarTitle} />
    )
    return (
      <Container topBar={topBar} isLoading={isLoading} style={{backgroundColor: colors.athensGray}} >
        <Modal
          animationType='fade'
          transparent
          visible={this.state.modalVisible} >
          <View style={{flex: 1, alignItems: 'center', backgroundColor: 'rgba(98, 97, 232, 0.85)'}}>
            <QuestionCard
              data={this.state.item}
              onPressWrite={this.onPressWrite}
              onPressSkip={this.onPressSkip} />
          </View>
        </Modal>
        { this.getAgesItems() }
      </Container>
    )
  }
}

const styles = StyleSheet.create({
  ageText: {
    textAlign: 'center',
    color: colors.governorBay,
    marginTop: 31,
    marginBottom: 17,
  },
})
