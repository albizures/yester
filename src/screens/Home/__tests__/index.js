import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Button, Alert } from 'react-native'

import agesFixture from './__fixtures__/ages.json'
import topicsFixture from './__fixtures__/topics.json'

import Container from '../../../components/Container'
import http from '../../../utils/http'
import Home from '../'
const navigation = {
  navigate: jest.fn(),
}

describe('src/screens/Home.js', () => {
  it('should render', () => {
    const tree = renderer.create(<Home navigation={navigation} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('when button is pressed', () => {
    it('should navigate to Home2 screen', () => {
      const shallowHome = shallow(<Home navigation={navigation} />)
      const shallowTouchableHighlights = shallowHome.find(Button)
      const shallowButtonToHome2 = shallowTouchableHighlights.at(0)

      shallowButtonToHome2.prop('onPress')()

      expect(navigation.navigate).toHaveBeenCalledTimes(1)
      expect(navigation.navigate).toHaveBeenCalledWith('Home2')
      expect(shallowHome.type()).toBe(Container)
    })
  })

  // shallowHome.setState({
  //   isLoading: false,
  //   dataSource: {
  //     age: agesFixture,
  //     topics: topicsFixture,
  //   },
  // })

  describe('#getAgesItems', () => {
    it('should render a View', () => {
      const age = agesFixture[0]
      const shallowHome = shallow(<Home navigation={navigation} />)
      const home = shallowHome.instance()

      shallowHome.setState({
        isLoading: false,
        age: agesFixture,
        topics: topicsFixture,
      })

      const shallowItem = shallow(home.getAgesItems({ item: age }))
      expect(shallowItem.type()).toBe('View')
    })
  })

  describe('#getTopicsItems', () => {
    it('should render a View', () => {
      const topic = topicsFixture[0]
      const shallowHome = shallow(<Home navigation={navigation} />)
      const home = shallowHome.instance()

      shallowHome.setState({
        isLoading: false,
        age: agesFixture,
        topics: topicsFixture,
      })

      const shallowItem = shallow(home.getTopicsItems({ item: topic }))
      expect(shallowItem.type()).toBe('View')
    })
  })

  describe('#renderAgeItem', () => {
    it('should render a View', () => {
      const age = agesFixture[0]
      const shallowHome = shallow(<Home navigation={navigation} />)
      const home = shallowHome.instance()

      shallowHome.setState({
        isLoading: false,
        age: agesFixture,
        topics: topicsFixture,
      })

      const shallowItem = shallow(home.renderAgeItem({ item: age }))
      expect(shallowItem.type()).toBe('View')
    })
  })

  describe('#renderTopicItem', () => {
    it('should render a View', () => {
      const topic = topicsFixture[0]
      const shallowHome = shallow(<Home navigation={navigation} />)
      const home = shallowHome.instance()

      shallowHome.setState({
        isLoading: false,
        age: agesFixture,
        topics: topicsFixture,
      })

      const shallowItem = shallow(home.renderTopicItem({ item: topic }))
      expect(shallowItem.type()).toBe('View')
    })
  })

  describe('#componentDidMount', () => {
    it('should render a View', async () => {
      http.get = jest.fn().mockResolvedValue({data: {value: 'test'}})
      const shallowHome = shallow(<Home navigation={navigation} />)
      const home = shallowHome.instance()

      await home.componentDidMount()
      expect(shallowHome.state()).toMatchObject({isLoading: false, value: 'test'})
    })

    describe('when the request fail', () => {
      it('should display an alert', async () => {
        http.get = jest.fn().mockRejectedValue(new Error('Error test'))
        Alert.alert = jest.fn()
        const shallowHome = shallow(<Home navigation={navigation} />)
        const home = shallowHome.instance()
        await home.componentDidMount()
        expect(Alert.alert).toBeCalledWith('Error', 'Error test')
      })
    })
  })
})
