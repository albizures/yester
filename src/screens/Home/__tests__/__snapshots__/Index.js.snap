// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/Home/index.js should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "marginTop": 40,
          "textAlign": "center",
        },
      ]
    }
  >
    HOME
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        to Home-2
      </Text>
    </View>
  </View>
  <View>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
  <ActivityIndicator
    animating={true}
    color="#000"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
        Object {
          "top": 20,
        },
      ]
    }
  />
</View>
`;
